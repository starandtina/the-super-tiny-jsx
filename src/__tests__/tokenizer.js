const tokenizer = require('../tokenizer')

it('tokenizer', () => {
  expect(
    tokenizer(`
      html {
        body {
          h1 {
            123456789
            "Hello World!"
          }
        }
      }
      `),
  ).toMatchSnapshot()
})
