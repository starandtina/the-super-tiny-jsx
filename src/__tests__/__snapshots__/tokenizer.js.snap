// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizer 1`] = `
Array [
  Object {
    "type": "ELEMENT",
    "value": "html",
  },
  Object {
    "type": "LEFT_CURLY_BRACKET",
    "value": "{",
  },
  Object {
    "type": "ELEMENT",
    "value": "body",
  },
  Object {
    "type": "LEFT_CURLY_BRACKET",
    "value": "{",
  },
  Object {
    "type": "ELEMENT",
    "value": "h1",
  },
  Object {
    "type": "LEFT_CURLY_BRACKET",
    "value": "{",
  },
  Object {
    "type": "NUMBER",
    "value": "123456789",
  },
  Object {
    "type": "STRING",
    "value": "Hello World!",
  },
  Object {
    "type": "RIGHT_CURLY_BRACKET",
    "value": "}",
  },
  Object {
    "type": "RIGHT_CURLY_BRACKET",
    "value": "}",
  },
  Object {
    "type": "RIGHT_CURLY_BRACKET",
    "value": "}",
  },
]
`;
